packages/nextcloud/test/helper.dart

```diff
@@ -127,12 +134,18 @@ Future<TestNextcloudClient> getTestClient(
     cookieJar: CookieJar(),
   );
 
+  var i = 0;
   while (true) {
-    // Test will timeout after 30s
     try {
       await client.core.getStatus();
       break;
-    } catch (_) {}
+    } on CoreApiException catch (error) {
+      i++;
+      await Future.delayed(const Duration(microseconds: 100));
+      if (i >= 3) {
```

Comment:
```suggestion
      await Future.delayed(const Duration(milliseconds: 100));
      if (i >= 30) {
```

This is 3 seconds in total. If there really is an infinite loop then that doesn't really matter, but we will have a lot less false positives.

Reply:
You might have noticed that I restricted the catch on dynamite exceptions.

This means that an error caused by false code (as was the problem before) won't even cause a loop anymore.

Reply:
Yes, but the container might still be slow to start. Apache is really bad with this, that's why most tests use php instead because the startup time is so much better.